sympy/solvers/solveset.py

```diff
@@ -978,6 +985,342 @@ def _solveset(f, symbol, domain, _check=False):
+        for mul_arg in Mul.make_args(add_arg):
+            if isinstance(mul_arg, (Pow, exp)) and \
+                    symbol in mul_arg.exp.free_symbols:
+                return True
+    return False
+
+
+def _transolve(f, symbol, domain):
+    r"""
+    Function to solve transcendental equations. It is a helper to
+    `solveset` and should be used internally as of now. `\_transolve`
+    currently supports the following class of equations:
+
+        - Exponential equations
+
+    Parameters
+    ==========
+
+    f : can be any transcendental equation that needs to be solved.
+        This needs to be an expression, which is assumed
+        to be equal to 0.
+
+    symbol : is variable for which the equation is solved.
+        This needs to be of class `Symbol`.
+
+    domain : is a set over which the equation is solved.
+        This needs to be of class `Set`.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import _transolve as transolve
+    >>> from sympy.solvers.solvers import _tsolve as tsolve
+    >>> from sympy import symbols, S, pprint
+    >>> x = symbols('x', real=True)
+    >>> transolve(5**(x - 3) - 3**(2*x + 1), x, S.Reals)
+    {-log(375)/(-log(5) + 2*log(3))}
+
+    How to use `\_transolve`
+    ========================
+
+    `\_transolve` should be used as an internal function. There should be
+    an internal call to `\_transolve`, passing all the three above
+    parameters as arguments. These three arguments should be sympy
+    instances.
+
+    How `\_transolve` works
+    =======================
+
+    `\_transolve` uses two types of helper functions to solve equation
+    of a particular class:
+
+    Identifying helpers: These helpers are used to determine whether a
+    given equation belongs to a certain class of equation or not.
+    These functions return either True or False.
+
+    Solving helpers: These are the helpers that solve the equation. They
+    are invoked once the equation is identified by the `identifying
+    helpers`. These function return either the complete solution or
+    modifies the equation for `solveset` to handle.
+
+    * Philosophy behind the module
+
+    `\_transolve` comes into action when `solveset` is unable to solve
+    the equation as a last resort to get the solutions. The idea is to
+    get the equation in a case (Add, Pow etc.) that satisfies its
+    general form, for example exponential equations take the form as
+    `a*f(x) + b*g(x)`, where `f(x)` and `g(x)` are power terms,
+    therefore they are included in `Add` case.
+
+    Once done different `identifying helper` conditions are included
+    to check which class the equation belongs. `Solving helper` is
+    invoked once the class of the equation is identified. If the
+    equation is solved the result is returned otherwise a
+    `ConditionSet` is returned.
+
+    How `\_transolve` is better than `\_tsolve`
+    ===========================================
+
+    1) Better output
```

Comment:
I would put this in the commit note. Once tsolve is gone this will have no significance.

Reply:
> Once tsolve is gone this will have no significance.

Yeah you are right but I guess until `transolve` completely overtakes `tsolve` we need to have comparisons (for developers to know why to use it).